### PR TITLE
[e2e] Refactor and document our extension of existing Playwright API and RFE features

### LINKED ISSUE #5523: [e2e] Refactor our extension of existing playwright APIs and playwright RFE APIs
#### Summary
We're nearing the point in our test framework implementation where we'll need to define how common functions should be exposed, used, and added to our test suite.

#### Tasks
- [x] Rename commonActions.js to appActions.js and keep in root folder. Add note about https://www.cypress.io/blog/2019/01/03/stop-using-page-objects-and-start-using-app-actions/
- [x] Renambe fixtures.js to _something which is obvious that it extends base fixtures. @ozyx to investigate. I think they use baseFixtures.js in their examples.  
- [x] Update documentation to be specific that only functions which extend the base playwright/test functionality go here. Any additional functions which are explicitly for those functions should be marked as private (codecoverage example).
- [x] Collect all examples of functions which extend the base functionality of playwright/test into fixtures.js. Off the top of my head, these functions are currently found in commonActions.js (waitForAnimationsToFinish) and visual test suites which manipulate the clock. 
- [x] Move all RFE APIs (APIs which existing a feature requests in the official playwright repo) should also be moved to a specific location like `./fixtures.js` or `./commonActions.js`
- [x] Document the above in our readme and within the file itself
- [x] Create a new function called 'getOrCreateDomainObject' for generating objects and returning the uuid
- [x] Update ts codehints and document heavily within this document and the readme

### PR DESCRIPTION
**Pull request by ozyx**
Closes #5523 #5436

### Describe your changes:

- Updates test:e2e:ci and all references to test:e2e:stable
- Remove firefox from unit testing to simply config
- Adds `appActions.js` as a home for app actions which are used across different tests and can be generalized
- Centralize all Playwright RFE features into `baseFixtures.js` and add documentation with GH issue references
  - Add custom fixture for overriding the browser clock
- Add `pluginFixtures.js` which extends `baseFixtures.js`
  - Add custom fixture which will create a domainObject of a certain type with the given name and default properties
- Lots of documentation

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Unit tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate unit tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Commit messages meet standards?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

File: e2e/tests/visual/generateVisualTestData.e2e.spec.js [hunk 1]
```diff
@@ -43,9 +42,6 @@ test('Generate Visual Test Data @localStorage', async ({ page, context }) => {
     // add overlay plot with defaults
     await page.locator('li:has-text("Overlay Plot")').click();
 
-    // Click on My Items in Tree. Workaround for https://github.com/nasa/openmct/issues/5184
```
Comment (unlikelyzero) [inline e2e/tests/visual/generateVisualTestData.e2e.spec.js:46]: cleanup

File: .circleci/config.yml [hunk 2]
```diff
@@ -94,30 +94,13 @@ jobs:
     parameters:
       node-version:
         type: string
-      browser:
-        type: string
     executor: pw-focal-development
     steps:
       - build_and_install:
           node-version: <<parameters.node-version>>
-      - when:
```
Comment (unlikelyzero) [inline .circleci/config.yml:103]: getting rid of useles paramaterization

File: e2e/playwright-ci.config.js [hunk 3]
```diff
@@ -4,12 +4,12 @@
 
 // eslint-disable-next-line no-unused-vars
 const { devices } = require('@playwright/test');
-const MAX_FAILURES = 5;
+const MAX_FAILURES = 0;
```
Comment (unlikelyzero) [inline e2e/playwright-ci.config.js:7]: this is just temporary as we address all @unstable tags

Comment (khalidadil) [reply e2e/playwright-ci.config.js:7]: @unlikelyzero does this need to stay in this PR?

File: e2e/tests/framework/exampleTemplate.e2e.spec.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+* 4. Any custom functions
+*
+*/
+
+//Structure: Some standard Imports. Please update the required pathing
+const { test, expect } = require('../../baseFixtures');
+const { createDomainObjectWithDefaults } = require('../../appActions');
+
+// Structure: Try to keep a single describe block per logical groups of tests. If your test runtime exceeds 5 minutes or 500 lines, it's likely that it will need to be split.
+// Annotations: Please use the @unstable tag so that our automation can pick it up as a part of our test promotion pipeline.
```
Comment (khalidadil) [inline e2e/tests/framework/exampleTemplate.e2e.spec.js:54]: @unlikelyzero does it make sense to default tests to unstable and mark stable ones with a "stable" flag? Right now we're asking devs/open source contributors to do this extra non-standard step that's the opposite of what was being done previously (specifying which test suites were stable).

Comment (unlikelyzero) [reply e2e/tests/framework/exampleTemplate.e2e.spec.js:54]: @khalidadil I think so, but I was trying to hit 5 birds with one stone.

We needed to tick the following boxes:
1. Provide a way to mark test and test suites as "approved for closed source execution".
2. Provide a way to mark new test and test suites as "new".
3. Provide a way to mark a test as generally flaky.
4. Mark individual tests inside of a larger test suite as unstable.
5. Get off of a string separated list inside of a npm run script.

I think we can both agree that stability, "closed source ready", and "new" tagging makes sense.

I think the pattern i'm promoting with `@unstable` assumes that we can move quickly in the open source and assume that all tests will eventually be executing without the `@unstable` tag. If we have 2000 tests (likely) it'll be easier to tag 5-10 of them as temporarily unstable.

We can always flip this in the future.

**Review by khalidadil — APPROVED** [review]

**Review by unlikelyzero — APPROVED** [review]

**Pull request merged**